cionic/api.py

```diff
@@ -374,21 +377,106 @@ def download_file(destpath: str, url: str, headers: dict = None) -> bool:
     return True
 
 
+def download_npz_from_metadata(
+    orgid: str,
+    studyid: str,
+    collection_num: int,
+    tokenpath: str,
+    outdir: str = '.',
+    overwrite: bool = False,
+    segmented: bool = False,
+    include_eulers: bool = True,
+    include_gait_splits: bool = True,
+) -> np.lib.npyio.NpzFile:
+    """
+    Downloads a .npz file associated with a specific org, study, and collection and
+    loads it. User friendly version.
+
+    Args:
+        orgid (str): Organization ID.
+        studyid (str): Short name of the study.
+        collection_num (int): Collection number within the study.
+        tokenpath (str): Path to the authentication token file. Must be specified.
+        outdir (str, optional): Output directory for the downloaded file.
+        overwrite (bool, optional): Whether to overwrite the file if it exists.
+        segmented (bool, optional): If True, loads and returns segmented data.
+        include_eulers (bool, optional): Whether to include Eulers in the download.
+        include_gait_splits (bool, optional): Whether to include gait splits in
+            the download.
+
+    Returns:
+        np.lib.npyio.NpzFile: Loaded .npz file.
+    """
+    if tokenpath is None:
+        raise ValueError("tokenpath must be specified to download NPZ from metadata.")
+
+    auth(tokenpath=tokenpath)
+    studies = get_cionic(f"{orgid}/studies")
+
+    sxid = None
+    for _, s in enumerate(studies):
+        if studyid == s['shortname']:
+            sxid = s['xid']
+
+    if sxid is None:
```

Comment:
instead of getting all of the studies for the org and then iterating through them, you can use the "friendly names" of the org and study and grab it directly. the API looks like this: 
`/c/v<version>/<orgname>/studies?name=<shortname>`

Reply:
not relevant anymore since you're using the `/collection` endpoint now